### ISSUE TITLE
Try to avoid the auto restart behaviour when resuming it

### ISSUE THREAD

**Issue by ylecollen** [vector-im/vector-android#119]
Many users reported us that the application is automatically restarted when it is resumed.

The memory warning is another case : the system warns us that there is no more available memory on the device and the client is going to be killed so it seems better to restart the client to reduce its memory usage and to avoid crash.

But there are other cases in which the application restarts whereas it is not expected.

With their help, we found out that the service (events polling thread) is killed or some other required fields are not initialised (e.g active sessions list) : the client detects that something is not valid in its fields and restarts to avoid crash.

I never got such behaviour on my devices (Nexus 6 - Android M, Samsung S3 - Android 4.2.2, Zenfone 550KL - android 5.01).

I suspect that there is a custom home launcher (or something similar) which kills service or stop the client in putting a pending activity to resume it.
I assume it appends when the device memory usage becomes high without being urgent.

**Comment by ylecollen** [comment]
By now, with the help of some users (thanks to them), we identified two use cases

1- The eventStreamService was released by the OS when the application is in background.
So, when the application is resumed, the activity might be created again (after a memory pressure) but the event stream is released : the client used to restart to manage this case

By now, the event stream is started again.

2- the application is resumed from the login activity.
It happens with some launchers or when the application activities are popped up by pressing the back button until to reach the device home page.

We already found out that Intent.FLAG_ACTIVITY_BROUGHT_TO_FRONT implies to display a temporary activity above existing one (it should be managed by now).
It seems there are other flags which might triggers invalid behaviours
